### PR TITLE
Add modifiers to Python bindings

### PR DESCRIPTION
**Pull request by rbberger**
## PR Summary

This adds the EOS modifiers to the Python bindings.

## PR Checklist

- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Format your changes by using the `make format` command after configuring with `cmake`.
- [x] Document any new features, update documentation for changes made.
- [x] Make sure the copyright notice on any files you modified is up to date.
- [x] After creating a pull request, note it in the CHANGELOG.md file

### CODE REVIEW

File: doc/sphinx/src/modifiers.rst [hunk 1]
```diff
@@ -103,14 +103,14 @@ for the specific enthalpy :math:`h`. This brings the sound speed formula into al
...
-parameters for the underlying EOS:
+for enthalpy by volume :math:`w`. The ``RelativisticEOS`` constructor accepts
+the underlying model, and the speed of light as parameter. For example, a
+relativistic ideal gas might be initialized as:
 
 .. code-block:: cpp
 
   using namespace singularity;
-  EOS my_eos = RelativisticEOS<IdealGas>(IdealGas(gm1, Cv));
+  EOS my_eos = RelativisticEOS<IdealGas>(IdealGas(gm1, Cv), cl);
```
Comment (Yurlungur) [inline doc/sphinx/src/modifiers.rst:114]: Oops good catch.

File: python/modifier_relativistic.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  eos_class<RelativisticEOS<T>>(m, std::string("Relativistic") + name)
+    .def(
+      py::init<T, Real>(),
+      py::arg("eos"), py::arg("cl")
+    );
+}
+
+template<typename T>
+void relativistic_eos_class(py::module_ & m, const char * name) {
+  // define Relativistic utility function
```
Comment (Yurlungur) [inline python/modifier_relativistic.cpp:35]: For developers we should document that these lists of template specializations need to be updated in tandem with the list in `eos.hpp`. Doesn't need to be now, but I wonder if we can automate that some how with template magic? @dholladay00 what are your thoughts?

File: singularity-eos/eos/modifiers/relativistic_eos.hpp [hunk 3]
```diff
@@ -56,6 +56,11 @@ class RelativisticEOS : public EosBase<RelativisticEOS<T>> {
   using EosBase<RelativisticEOS<T>>::PTofRE;
   using EosBase<RelativisticEOS<T>>::FillEos;
 
+  // give me std::format or fmt::format...
+  static std::string EosType() {
+    return std::string("RelativisticEOS<") + T::EosType() + std::string(">");
+  }
```
Comment (Yurlungur) [inline singularity-eos/eos/modifiers/relativistic_eos.hpp:62]: This is nice

**Review by Yurlungur — APPROVED** [review]
This is great! My one comment here would be that the modifiers introduce a lot of boiler plate, which it would be nice to automate away. But that can be for another time.

**Comment by Yurlungur** [comment]
Merging now, since I'm happy with this as-is, but @dholladay00 I'd like your thoughts on type automaticion.

File: python/modifier_scaled.cpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  eos_class<ScaledEOS<ShiftedEOS<T>>>(m, std::string("ScaledShifted") + name)
+      .def(py::init<ShiftedEOS<T>, Real>(), py::arg("eos"), py::arg("scale"));
+
+  // define scaled class
+  eos_class<ScaledEOS<T>>(m, std::string("Scaled") + name)
+      .def(py::init<T, Real>(), py::arg("eos"), py::arg("scale"));
+}
+
+void create_scaled_eos_classes(py::module_ &m) {
+  scaled_eos_class<IdealGas>(m, "IdealGas");
```
Comment (dholladay00) [inline python/modifier_scaled.cpp:39]: I'm curious if we could write this as:
```c++
template<typename ... ARGS>
void create_scaled_eos_classes_var(py::module_ &m, tl<ARGS...>) {
  scaled_eos_class<ARG>(m, typeid(decltype(ARG)).name().c_str())...;
}
```
then `create_scaled_eos_classes_var(m, full_eos_list);` where `full_eos_list` is defined in eos.hpp. There may need to be some playing with the `,` operator to make it work, not sure, but something like that should be workable.

Comment (Yurlungur) [reply python/modifier_scaled.cpp:39]: This would be a nice next step. @rbberger ?

**Pull request merged**